testing/trino-tests/src/test/java/io/trino/execution/TestQueues.java

```diff
@@ -300,6 +300,20 @@ public void testSelectorResourceEstimateBasedSelection()
         }
     }
 
+    @Test
+    @Timeout(10)
+    public void testQueryTextBasedSelection()
+            throws Exception
+    {
+        try (QueryRunner queryRunner = TpchQueryRunner.builder().build()) {
+            queryRunner.installPlugin(new ResourceGroupManagerPlugin());
```

Comment:
on my rather fast laptop this test took 4.3s
the 10 s timeout feels small, esp given the test creates a new QR

- https://github.com/trinodb/trino/issues/27611